src/modules/order.ts

```diff
@@ -63,6 +82,18 @@ export class OrderModule extends BaseModule {
       };
     });
 
+    if (params[0].setLeverage) {
+      const leverage = params[0].setLeverage
+      msgs.unshift({
+        typeUrl: CarbonTx.Types.MsgSetLeverage,
+        value: {
+          creator: wallet.bech32Address,
+          market: params[0].market,
+          leverage: leverage.shiftedBy(18).toString(10),
+        },
+      })
+    }
```

Comment:
for TPSL orders. on the front end, setLeverage is set for the very first order, before the TP/SL order is added into the array. is this robust? initially checked every iteration in the `.map()` (previous commit).

will need to document behaviour on frontend

Reply:
maybe we can have `params: (OrderModule.CreateOrderParams | OrderModule.SetLeverageParams) [], …` instead

Reply:
so for a single order we add leverage into the orderParams, but for TPSL / multi order we submit an array of orders + leading setLeverage ?

Reply:
think should be fine